model/README.md

```diff
@@ -0,0 +1,7 @@
+# Model
+
+Package `model` provides modeling components, such as domains and ranges.
```

Comment:
Maybe it's worth noting the use-case of these? To me it seems they are most useful when doing constraint programming and propagations.